### ISSUE TITLE
parser scalacheck tests sometimes fail

### ISSUE THREAD

**Issue by johnynek** [johnynek/bosatsu#35]
```
[info] - we can parse quoted strings *** FAILED *** (200 milliseconds)
[info]   TestFailedException was thrown during property evaluation.
[info]     Message: failed to parse: : "\u0000" at 0 in region empty string, idx = 0 with trace: escapedString:1:1 / "\u0000":1:1 ...""
[info]     Location: (ParserTest.scala:40)
[info]     Occurred when passed generated values (
[info]       arg0 = (,,) // 18 shrinks
[info]     )
```

This may be a corner case of generation, and possibly a bad example due to bad (or no) shrink instances.

**Comment by johnynek** [comment]
I think this was fixed by #86

**Comment by johnynek** [comment]
This is still there:

```
[info] - we can parse quoted strings *** FAILED *** (287 milliseconds)
[info]   TestFailedException was thrown during property evaluation.
[info]     Message: failed to parse: : "\u0000" at 0 in region empty string, idx = 0 with trace: escapedString:1:1 / "\u0000":1:1 ...""
[info]     Location: (ParserTest.scala:48)
[info]     Occurred when passed generated values (
[info]       arg0 = (,,) // 20 shrinks
[info]     )
```
seen at: https://travis-ci.org/johnynek/bosatsu/jobs/447261924

Probably a real bug and and an example of shrinking going wrong. Turning up the the number of iterations to reliably trigger it would be good.
